test/includes.spec.js

```diff
@@ -1,95 +1,199 @@
+            return server.injectThen({method: 'get', url: '/people?include=soulmate'}).then(function (res) {
+                expect(res.statusCode).to.equal(200);
+                const body = res.result;
+                expect(body.data).to.have.length(2);
+                expect(body).to.have.property('included');
+                expect(body.included).to.be.an.Array;
+                expect(body.included).to.have.length(1);
+                expect(body.included[0]).to.have.property('id', 'c344d722-b7f9-49dd-9842-f0a375f7dfdc');
+                expect(body.included[0]).to.have.property('type', 'people');
+            });
+        });
+        it('should include soulmate when getting resource by id', function () {
+            return server.injectThen({method: 'get', url: '/people/abcdefff-b7f9-49dd-9842-f0a375f7dfdc?include=soulmate'}).then(function (res) {
+                expect(res.statusCode).to.equal(200);
+                const body = res.result;
+                expect(body.data).to.be.an.Object;
+                expect(body).to.have.property('included');
+                expect(body.included).to.be.an.Array;
+                expect(body.included).to.have.length(1);
+                expect(body.included[0]).to.have.property('id', 'c344d722-b7f9-49dd-9842-f0a375f7dfdc');
+                expect(body.included[0]).to.have.property('type', 'people');
+            });
+        });
+    });
+
+    describe('compound documents', function () {
+        it('should include pet and person when querying collars', function () {
+            return server.injectThen({method: 'get', url: '/collars?include=collarOwner.owner.soulmate,collarOwner,collarOwner.owner'})
+                .then(function (res) {
+                    const body = res.result;
+                    expect(body.included).to.be.an.Array;
+                    expect(body.included).to.have.length(3);
+                    _.forEach(body.included, function (item) {
+                        if (item.type === 'pets' && item.id === 'b344d722-b7f9-49dd-9842-f0a375f7dfdc') {
+                            return;
+                        }
+                        if (item.type === 'people' && item.id === 'abcdefff-b7f9-49dd-9842-f0a375f7dfdc') {
+                            return;
+                        }
+                        if (item.type === 'people' && item.id === 'c344d722-b7f9-49dd-9842-f0a375f7dfdc') {
+                            return;
+                        }
+                        throw new Error('Unexpected included item: ' + JSON.stringify(item, null, 2));
+                    });
+                });
+        });
+        describe('when relationship not defined on schema', function () {
+            it('should respond with 500', function () {
+                return server.injectThen({method: 'get', url: '/ents?include=owner'})
+                    .then(function (res) {
+                        expect(res.statusCode).to.equal(500);
```

Comment:
Such situation should not take place, but I wanted to cover proper error handling (test if app will hang or not).